src/AmplNLWriter.jl

```diff
@@ -451,33 +456,91 @@ function add_to_index_maps!(forward_map::Dict{Int64, Int64},
 end
 
 function read_results(m::AmplNLMathProgModel)
+    did_read_solution = read_sol(m)
+
+    # Convert solve_result
+    if 0 <= m.solve_result_num < 100
+        m.status = :Optimal
+        m.solve_result = "solved"
+    elseif 100 <= m.solve_result_num < 200
+        # Used to indicate solution present but likely incorrect.
+        m.status = :Optimal  # TODO maybe something else
+        m.solve_result = "solved?"
```

Comment:
Maybe this should have something like:
`warn("AMPL has indicated that the solver has returned the status :Optimal, but that there might be an error in the solution. The message from the solver was: $(m.solve_message)")`